client/lib/signup/step-actions/index.js

```diff
@@ -188,6 +196,10 @@ export const getNewSiteParams = ( {
 			timezone_string: guessTimezone(),
 			wpcom_public_coming_soon: launchAsComingSoon,
 			...( siteAccentColor && { site_accent_color: siteAccentColor } ),
+			...( flowToCheck === 'guided' && setDevAccount && { set_dev_account: setDevAccount } ),
+			...( flowToCheck === 'guided' && siteGoals && { site_goals: siteGoals } ),
+			...( flowToCheck === 'guided' &&
+				onboardingSegment && { onboarding_segment: onboardingSegment } ),
```

Comment:
Let's make an object called `guidedFlowProps` and populate if flow is guided. 

```js
const guidedFlowProps = {};
if ( flowToCheck === 'guided' ) {
	guidedFlowProps.set_dev_account = setDevAccount;
    .
    .
}
```

Reply:
Updated